Fix passphrase conversion when coming from legacy wallets

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->

#1483 

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- 0a72efc8a84a700d3d524c3c9883b35bd3bf8eb5
  :round_pushpin: **only CBOR-serialize the passphrase when checking its hash**
  The actual encryption passphrase that is used is the hash provided to the API
by clients, it can be either null or a 32-byte string. Then, this passphrase
is serialized to CBOR before being hashed, but, that is the non-serialized
passphrase that is used for encrypted the root key!

- 6c88eead09e29614a86d235b79b9e2b177695b7f
  :round_pushpin: **use golden string generated from cardano-sl implementation!**
  The previous strings were generated from our implementation, which disqualifies them from being valid
test cases. When comparing with a legacy implementation, the data MUST come from the legacy implementation
otherwise the test is pointless.



# Comments

<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

KtorZ committed:
only CBOR-serialize the passphrase when checking its hash

The actual encryption passphrase that is used is the hash provided to the API
by clients, it can be either null or a 32-byte string. Then, this passphrase
is serialized to CBOR before being hashed, but, that is the non-serialized
passphrase that is used for encrypted the root key!

paweljakubas commented on lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs:
```diff
@@ -527,21 +527,13 @@ encryptPassphrase  (Passphrase bytes) = do
 -- | Manipulation done on legacy passphrases before getting encrypted.
---
--- Yes there's CBOR. Yes it's hashed before being encrypted. You don't actually
--- pronounce the 'l' in salmon. A lot of things in the world don't make sense
```
I will miss this comment ;-)

KtorZ replied:
Me too. It'll live in the git history and will be our little secret 🤫

KtorZ committed:
use golden string generated from cardano-sl implementation!

The previous strings were generated from our implementation, which disqualifies them from being valid
test cases. When comparing with a legacy implementation, the data MUST come from the legacy implementation
otherwise the test is pointless.

KtorZ committed:
re-generate nix machinery
